src/main/java/org/radarcns/management/service/UserService.java

```diff
@@ -351,10 +355,22 @@ public User getUserWithAuthorities() {
     public void removeNotActivatedUsers() {
         log.info("Scheduled scan for expired user accounts starting now");
         ZonedDateTime cutoff = ZonedDateTime.now().minus(Period.ofDays(3));
+
+        // first delete non-activated users related to subjects
+        userRepository.findAllByActivated(false).stream()
+                .filter(user -> revisionService.getAuditInfo(user).getCreatedAt().isBefore(cutoff))
```

Comment:
Can't we query once instead of querying twice?

Reply:
I was wondering, what if we trigger cleaning of subjects, which deletes associated user instead of user deleting subject. I was reading a book on clean architecture, and it was mentioned that the inner circle should not know/depend on outer circle. In this case, the subject should/can know about user, but not the other way around. What do you think @dennyverbeeck @blootsvoets